src/common/obj_bencher.cc

```diff
@@ -388,10 +388,18 @@ int ObjBencher::write_bench(int secondsToRun,
     }
     lock.Unlock();
     //create new contents and name on the heap, and fill them
-    newContents = new bufferlist();
     newName = generate_object_name(data.started);
-    snprintf(data.object_contents, data.object_size, "I'm the %16dth object!", data.started);
-    newContents->append(data.object_contents, data.object_size);
+    newContents = contents[slot];
+    if (!newContents) {
+        newContents = new bufferlist();
+        contents[slot] = newContents;
+        snprintf(data.object_contents, data.object_size, "I'm the %16dth object!", data.started);
+        newContents->append(data.object_contents, data.object_size);
+    } else {
+        // hacky way to invalidate internal crc cache
+        newContents->zero(0, 0);
+        snprintf(newContents->c_str(), data.object_size, "I'm the %16dth object!", data.started);
+    }
```

Comment:
What about using bufferlist::copy_in instead of the zero() + snprintf combo ? It looks like marginaly more expensive but more maintainable. Does it show a visible difference when you test it ?